cmake/catkin_package.cmake

```diff
@@ -286,7 +286,8 @@ function(_catkin_package)
   set(DEVELSPACE TRUE)
   set(INSTALLSPACE FALSE)
 
-  set(PROJECT_SPACE_DIR ${CATKIN_DEVEL_PREFIX})
+  set(PROJECT_SPACE_LIBDIR ${CATKIN_DEVEL_PREFIX}/lib)
+  set(PROJECT_SPACE_DATADIR ${CATKIN_DEVEL_PREFIX}/share)
```

Comment:
These intentionally _do not_ use the GNUInstallDirs vars, for two reasons:

- The develspace is not a "real" FHS tree and has its own internal standard that shouldn't be impacted by whatever overrides come in from the CMAKE_INSTALL_XXXX variables.
- If the GNUInstallDirs vars are set to absolute paths (as in the Nix build case), those absolute paths will be appended to the devel prefix, which is almost certainly not what is wanted here.

We could always check for absolute paths and only append in the case where they aren't, but then what to do when they are? I don't think adding branchiness here is worth whatever gains there might be, particularly when the develspace is dropped in ROS 2 anyway.